### ISSUE TITLE
Nuxt 3.9.0 Internal server error: Soft-invalidated module "entry.js" should not have existing transform result

### ISSUE THREAD

**Issue by serkodev** [unocss/unocss#3472]
### UnoCSS version

0.58.2

### Describe the bug

After running nuxt dev server, go to browser and refresh the page once, the error will show up.
```
Internal server error: Soft-invalidated module "/node_modules/nuxt/dist/app/entry.js" should not have existing transform result
```

### Reproduction

1. create a new nuxt project
2. add `@unocss/nuxt` module
3. run `nuxt dev`
4. go to browser, no error on first load
5. (Require this step) refresh the page once and the error shows up

https://stackblitz.com/edit/github-xtxzaj?file=package.json

### Additional info
Already try to add the following config regarding to https://github.com/unocss/unocss/issues/3468#issuecomment-1871049463 but still having error
```ts
export default defineConfig({
  features: {
    inlineStyles: false,
  },
})
```

### System Info

```
"nuxt": "^3.9.0",
"vue": "^3.4.0",
"vue-router": "^4.2.5"
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/unocss/unocss/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't [already an issue](https://github.com/unocss/unocss/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/unocss/unocss/discussions) or join our [Discord Chat Server](https://chat.antfu.me).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by serkodev** [comment]
update: after testing with the latest unocss (0.58.3), this issue is still not solved

**Comment by kissu** [comment]
I do indeed have the same issue but it kinda disappears after a few refreshes? Not really consistent but not a huge blocker either. 🤔

**Comment by cloydlau** [comment]
+1, I have temporarily downgraded nuxt to version 3.8.2.

**Comment by gigantino** [comment]
+1, having the same issue.

**Comment by jandobrx** [comment]
Also having this issue 

> I do indeed have the same issue but it kinda disappears after a few refreshes? Not really consistent but not a huge blocker either. 🤔

But this does seem to do the trick as-well no issues with my build and deployments! But big red internal server error sends gives me chills 🙃

**Comment by kissu** [comment]
Working with Netlify Edge functions locally is not always perfect (works fine on an M3 chip but less on a Linux machine for some reason) but still achievable, just need to guess that the server crashes before the functions are called and restart Netlify CLI.

**Comment by kstraszewski** [comment]
After upgrading to Nuxt 3.9.3 everything is working fine for me.

**Comment by ironytr** [comment]
unocss 5.8.3
nuxt 3.9.3 
`Internal server error: Soft-invalidated module "/node_modules/.pnpm/nuxt@3.9.3_eslint@8.56.0_rollup@4.9.5_sass@1.70.0_typescript@5.3.3_vite@5.0.12/node_modules/nuxt/dist/app/entry.js" should not have existing transform result`

**Comment by dellumdeus** [comment]
Any news on this?

**Comment by AndrewBogdanovTSS** [comment]
unocss 5.8.3
@unocss/nuxt 0.58.3
nuxt 3.9.3
vue 3.4.15
![image](https://github.com/unocss/unocss/assets/2766448/a5339d39-8119-4cf7-92ce-8e6388679daa)

**Comment by owl1n** [comment]
@antfu any updates?

The error is sometimes so intrusive that it is impossible to work. Sometimes, 1-2 page refreshes helps, but sometimes 40 refreshes still with error

**Comment by gigantino** [comment]
TL;DR put this in your Nuxt config:

```ts
export default defineConfig({
  features: {
    inlineStyles: false,
  },
})
```

[Original answer](https://github.com/unocss/unocss/issues/3468#issuecomment-1871049463)

**Comment by daniandl** [comment]
> TL;DR put this in your Nuxt config:
> 
> ```ts
> export default defineConfig({
>   features: {
>     inlineStyles: false,
>   },
> })
> ```
> 
> [Original answer](https://github.com/unocss/unocss/issues/3468#issuecomment-1871049463)

The error still appears despite this

**Comment by antfu** [comment]
I can't reproduce with @serkodev's reproduction with the latest versions anymore. Can anyone provide another minimal reproductions?

**Comment by rylanharper** [comment]
Hmm I am still seeing the bug in the latest release, version `^0.58.4`. The issue arises during the initial startup but is resolved upon refreshing the page.. Although its hard to replicate, after that first initial issue, I can't seem to get it to reappear when I start/stop the dev sever again.

^^ Actually I have to wait about 30sec to a min after stopping the dev sever for it to reappear. Here is a screenshot from just right now:

<img width="2560" alt="Screenshot 2024-01-25 at 10 40 31 AM" src="https://github.com/unocss/unocss/assets/16216160/70192809-c9da-4d1b-9460-6ad9feb3be66">

**Comment by naknakLEE** [comment]
+1, having the same issue.

**Comment by antfu** [comment]
Again, I would need a minimal reproduction to investigate this - saying +1 does not improve the situation, for the record. Thanks

**Comment by re-ovo** [comment]
Under my test, I need to open F12, disable network cache, it only seems to appear on the second load, if the cache is not disabled, then it will not load the second time

**Comment by hunghg255** [comment]
I'm not coding nuxt, but I tried debug error after I read on Twitter =)))
I recorded an error in this link, everybody can check:  https://streamable.com/hvrvnw

Step: 
1: npm run dev open localhost at the first browser 
2: turn off npm run dev
3: remove .nuxt and node_modules/.cache
4: open 2nd browser
5: npm run dev
=> The error will appear on the first browser

**Comment by kissu** [comment]
> I'm not coding nuxt, but I tried debug error after I read on Twitter =))) I recorded an error in this link, everybody can check: https://streamable.com/hvrvnw
> 
> Step: 1: npm run dev open localhost at the first browser 2: turn off npm run dev 3: remove .nuxt and node_modules/.cache 4: open 2nd browser 5: npm run dev => The error will appear on the first browser

Do you please have the link for your code? À video record is not considered a valid reproduction unfortunately. 😅

**Comment by huangyan321** [comment]
https://stackblitz.com/edit/github-dqvhqs?file=package.json
This link provides the minimal reproduction. To reproduce the warning, please follow these steps:

1. Wait for the Nuxt build to complete.
2. Once the page has finished loading, click refresh once.

**Comment by deglier** [comment]
I have the same issue:

nuxt:  3.9.3
vue: 3.4.14
vue-router: 4.2.5
@unocs/nuxt: 0.58.3

When I start the local server with the script: ``nuxt dev`` and open it in the browser, the error does not occur. But when I refresh the page, the error appears and then disappears.

When I reinstall everything, including cleaning the pnpm lock file, the server becomes stable and error-free until I restart the PC.

![image](https://github.com/unocss/unocss/assets/13968200/fa47d60c-b578-4205-a70a-357a99fd15b2)

**Comment by owl1n** [comment]
> Again, I would need a minimal reproduction to investigate this - saying +1 does not improve the situation, for the record. Thanks

You can use any one reproduction from above and test it like just refresh the page in browser. I think its not working on stackbliz or any sandboxes, you need start it on local environment

**Comment by NMTuan** [comment]
> I can't reproduce with @serkodev's reproduction with the latest versions anymore. Can anyone provide another minimal reproductions?我无法再使用最新版本进行重现。谁能提供另一个最小的复制品？

try it? https://github.com/NMTuan/nuxt_393_error

**Comment by kissu** [comment]
Maybe it is also depending on the environment. In my case Linux (Ubuntu) works worse than MacOS hence it's a bit tricky to reproduce properly on MacOS.

**Comment by serkodev** [comment]
Already update the reproduction link with the latest UnoCSS version `0.58.4` and the error still shows up.
https://stackblitz.com/edit/github-xtxzaj?file=package.json

```
"@unocss/nuxt": "^0.58.4",
"nuxt": "^3.9.0",
"vue": "^3.4.0",
"vue-router": "^4.2.5"
```
### Steps
1. go to browser and wait for loading complete, there's no error on first load
2. (Require this step) refresh the page once and the error shows up

**Comment by antfu** [comment]
It's probably introduced in https://github.com/vitejs/vite/pull/14654, not yet sure how to fix it, investigating

**Comment by antfu** [comment]
Alright, hopefully https://github.com/vitejs/vite/pull/15768 should fix this

**Comment by serkodev** [comment]
Just tested and this issue should be fixed **since** `vite v5.1.0` (`nuxt v3.10.2`).
Related commits: https://github.com/vitejs/vite/pull/15768, https://github.com/vitejs/vite/pull/15785

Thank you @antfu, @bluwy 🍻

**Comment by danielwaltz** [comment]
I wonder, now that the original issue has been fixed upstream in Vite, would we be able to remove or update the default value of the `unocss.disableNuxtInlineStyle` option?
